dnf/module/module_base.py

```diff
@@ -140,31 +140,139 @@ def install(self, module_specs, strict=True):
+                                    [profile.getName() for profile in available_profiles]))
+                                msg = _("Unable to match profile for argument {}. Available "
+                                        "profiles for '{}:{}': {}").format(
+                                    spec, name, stream, profile_names)
+                            else:
+                                msg = _("Unable to match profile for argument {}").format(spec)
+                            logger.error(msg)
+                            no_match_specs.append(spec)
+                            continue
+                    elif name in removed_profiles:
+
+                        for profile in removed_profiles[name]:
+                            module_profiles = latest_module.getProfiles(profile)
+                            if not module_profiles:
+                                logger.warning(
+                                    _("Installed profile '{0}' is not available in module "
+                                      "'{1}' stream '{2}'").format(profile, name, stream))
+                                continue
+                            profiles.extend(module_profiles)
+                    for profile in profiles:
+                        self.base._moduleContainer.install(latest_module, profile.getName())
+                        for pkg_name in profile.getContent():
+                            install_dict.setdefault(pkg_name, set()).add(spec)
+                    for module in install_module_list:
+                        artifacts = module.getArtifacts()
+                        install_set_artefacts.update(artifacts)
+                        for artifact in artifacts:
+                            arch = artifact.rsplit(".", 1)[1]
+                            if arch in src_arches:
+                                continue
+                            pkg_name = artifact.rsplit("-", 2)[0]
+                            new_artefacts_names.add(pkg_name)
+        if fail_safe_repo_used:
+            raise dnf.exceptions.Error(_(
+                "Installing module from Fail-Safe repository is not allowed"))
+        install_base_query, profiles_errors = self._install_profiles_internal(
+            install_set_artefacts, install_dict, strict)
+        if profiles_errors:
+            error_specs.extend(profiles_errors)
+
+        # distrosync module name
+        all_names = set()
+        all_names.update(new_artefacts_names)
+        all_names.update(active_artefacts_names)
+        remove_query = self.base.sack.query().filterm(empty=True)
+        for pkg_name in all_names:
+            query = self.base.sack.query().filterm(name=pkg_name)
+            installed = query.installed()
+            if not installed:
+                continue
+            available = query.available()
+            if not available:
+                logger.warning(_("No packages available for distribution synchronization"
```

Comment:
Do you mean "distro-sync" by "distribution synchronization"? If yes, it's hard to make the connection. I'm afraid this message will be very confusing for unexperienced users. But not sure what to use tbh. Maybe "for installation" would be enough? Or "for distro-sync installation"?

Reply:
Good point, but `installation` is not the correct word. What about to use `distrosync`. We use it in documentation.

Reply:
Okay, sounds good.